src/AnalysisDD4hep.cxx

```diff
@@ -0,0 +1,735 @@
+  TTreeReaderArray<float> HcalEndcapNClusters_x(tr,      "HcalEndcapNClusters.position.x");
+  TTreeReaderArray<float> HcalEndcapNClusters_y(tr,      "HcalEndcapNClusters.position.y");
+  TTreeReaderArray<float> HcalEndcapNClusters_z(tr,      "HcalEndcapNClusters.position.z");
+  TTreeReaderArray<float> HcalEndcapNClusters_theta(tr,  "HcalEndcapNClustersInfo.polar.theta");
+  TTreeReaderArray<float> HcalEndcapNClusters_phi(tr,    "HcalEndcapNClustersInfo.polar.phi");
+
+  //HcalBarrel
+  TTreeReaderArray<float> HcalBarrelClusters_energy(tr, "HcalBarrelClusters.energy");
+  TTreeReaderArray<float> HcalBarrelClusters_x(tr,      "HcalBarrelClusters.position.x");
+  TTreeReaderArray<float> HcalBarrelClusters_y(tr,      "HcalBarrelClusters.position.y");
+  TTreeReaderArray<float> HcalBarrelClusters_z(tr,      "HcalBarrelClusters.position.z");
+  TTreeReaderArray<float> HcalBarrelClusters_theta(tr,  "HcalBarrelClustersInfo.polar.theta");
+  TTreeReaderArray<float> HcalBarrelClusters_phi(tr,    "HcalBarrelClustersInfo.polar.phi");
+
+  //Ecal
+  TTreeReaderArray<float> EcalEndcapPClusters_energy(tr, "EcalEndcapPClusters.energy");
+  TTreeReaderArray<float> EcalEndcapPClusters_x(tr,      "EcalEndcapPClusters.position.x");
+  TTreeReaderArray<float> EcalEndcapPClusters_y(tr,      "EcalEndcapPClusters.position.y");
+  TTreeReaderArray<float> EcalEndcapPClusters_z(tr,      "EcalEndcapPClusters.position.z");
+  TTreeReaderArray<float> EcalEndcapPClusters_theta(tr,  "EcalEndcapPClustersInfo.polar.theta");
+  TTreeReaderArray<float> EcalEndcapPClusters_phi(tr,    "EcalEndcapPClustersInfo.polar.phi");
+
+  TTreeReaderArray<float> EcalEndcapNClusters_energy(tr, "EcalEndcapNClusters.energy");
+  TTreeReaderArray<float> EcalEndcapNClusters_x(tr,      "EcalEndcapNClusters.position.x");
+  TTreeReaderArray<float> EcalEndcapNClusters_y(tr,      "EcalEndcapNClusters.position.y");
+  TTreeReaderArray<float> EcalEndcapNClusters_z(tr,      "EcalEndcapNClusters.position.z");
+  //watch out the branch name
+  TTreeReaderArray<float>  EcalEndcapNClusters_theta(tr,  "EcalEndcapNClusterInfo.polar.theta");
+  TTreeReaderArray<float>  EcalEndcapNClusters_phi(tr,    "EcalEndcapNClusterInfo.polar.phi");
+
+  TTreeReaderArray<float> EcalBarrelClusters_energy(tr, "EcalBarrelImagingClusters.energy");
+  TTreeReaderArray<float> EcalBarrelClusters_x(tr,      "EcalBarrelImagingClusters.position.x");
+  TTreeReaderArray<float> EcalBarrelClusters_y(tr,      "EcalBarrelImagingClusters.position.y");
+  TTreeReaderArray<float> EcalBarrelClusters_z(tr,      "EcalBarrelImagingClusters.position.z");
+  TTreeReaderArray<float> EcalBarrelClusters_theta(tr,  "EcalBarrelImagingClustersInfo.polar.theta");
+  TTreeReaderArray<float>  EcalBarrelClusters_phi(tr,    "EcalBarrelImagingClustersInfo.polar.phi");
+
+  TTreeReader::EEntryStatus entrystats = tr.SetEntry(0);
+
+  // open output file
+  // FIXME: replace name variable
+  outFile = new TFile(outfilePrefix,"RECREATE");
```

Comment:
we can figure out how to better use inheritance between the two Analysis classes, so for example the output file name is created in one place; I can help with this when we refactor `histSet` (see https://github.com/c-dilks/largex-eic/pull/7)